src/plugins/layout/stateprovider.ts

```diff
@@ -99,30 +94,6 @@ export function createElementLayoutState(element: Element, forceMask: boolean):
   return elementState;
 }
 
-export function createStyleLayoutState(styleNode: HTMLStyleElement, forceMask: boolean): IStyleLayoutState {
-  let layoutState = createElementLayoutState(styleNode, forceMask) as IStyleLayoutState;
-  if (config.cssRules) {
-    let cssRules = layoutState.cssRules = null;
-
-    // Firefox throws a SecurityError when trying to access cssRules of a stylesheet from a different domain
-    try {
-      let sheet = styleNode.sheet as CSSStyleSheet;
-      cssRules = sheet ? sheet.cssRules : [];
-    } catch (e) {
-      if (e.name !== "SecurityError") {
-        throw e;
-      }
-    }
-    if (cssRules !== null) {
-      layoutState.cssRules = [];
-      for (let i = 0; i < cssRules.length; i++) {
-        layoutState.cssRules.push(cssRules[i].cssText);
-      }
-    }
-  }
-  return layoutState;
-}
```

Comment:
Somewhat redundant comment, but why remove this? If we need to compute cssRules (as an extra vs. default element node), we should have a function for it here. You can define the cssRules function in this file, so that it can be exported to layout.ts as well

Reply:
Fixed

---
In reply to: [243079814](https://github.com/Microsoft/clarity-js/pull/116#discussion_r243079814) [](ancestors = 243079814)